fix: suggestions for rules additional to Airbnb

andimeier-ch opened the pull request.

Commit:
fix: suggestions for rules additional to Airbnb

Commit:
fix(rules): remove unused exemption rules

fxstr commented on rules/joinbox-custom.js:
```diff
@@ -1,47 +1,13 @@
+        'import/extensions': ['error', 'always', { ignorePackages: true }],
+
+        // do not require empty lines between single line class members, such as private
+        // properties which are defined at the beginning of the class.
```
Evtl. Reason erwähnen? Insb., dass Private Props in Airbnb nicht explizit erwähnt werden?

fxstr commented on rules/joinbox-custom.js:
```diff
@@ -1,47 +1,13 @@
-        // warning for no-param-reassign
-        'no-param-reassign': 'warn',
+        // requires that file extentions are used in import statements, exept for package names
+        'import/extensions': ['error', 'always', { ignorePackages: true }],
```
Ah, jetzt erst verstanden – ich würde die Zeile ganz streichen: `.js` oder `.mjs` ist eigentlich ein Implementation Detail, das nicht relevant sein sollte beim Einbinden. Oder?

fxstr reviewed (commented):
👏

Commit:
build: update packages and adjust test file

- update eslint
- update eslint-config-airbnb-base
- add @babel/eslint-parser and respective configs
- adjust test file to our new rules

fxstr commented on package.json:
```diff
@@ -14,13 +14,13 @@
   ],
   "author": "Lorenzo Kappeler",
```
Passisch mer dä au no hurti aa?

Commit:
refactor: adjust author name in package.json
